cpp/src/bitmask/legacy/bitmask_ops.cu

```diff
@@ -163,9 +163,21 @@ gdf_error gdf_count_nonzero_mask(gdf_valid_type const* masks,
 
 gdf_error gdf_mask_concat(gdf_valid_type* output_mask,
                           gdf_size_type output_column_length,
-                          gdf_valid_type* masks_to_concat[],
-                          gdf_size_type* column_lengths,
+                          gdf_column *columns_to_concat[],
                           gdf_size_type num_columns) {
+  std::vector<gdf_valid_type*> h_masks(num_columns);
+  std::vector<gdf_size_type> h_column_lengths(num_columns);
+  std::generate(h_masks.begin(), h_masks.end(),
```

Comment:
I think it would be a little simpler to use a `std::transform` something like this:
```
std::transform(columns_to_concat, columns_to_concat + num_columns, h_masks.begin(),
                       [](gdf_column* col){return col->valid;});
```

Reply:
Yeah, forgot you can use pointers as iterators...